### PR TITLE
feat: Add projection on inserted_at

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

Since we added `inserted_at` batch exports are querying events based on `inserted_at`.

However, we are forced to include `timestamp` in the query to better utilize the sort key. It would be better if we could use a sort key based on `inserted_at` to optimize performance. 

Moreover, due to the discrepancy between `inserted_at` and `timestamp` we are missing events that were historically loaded.

## Changes

Add a migration that creates a projection with `inserted_at` in the sort key.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/clickhouse/migrations/0049_add_inserted_at_projection.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ON CLUSTER '{cluster}'
+ADD PROJECTION events_inserted_at_projection (
+  SELECT * ORDER BY (team_id, inserted_at, event, cityHash64(distinct_id), cityHash64(uuid))
+);
+"""
+
+MATERIALIZE_PROJECTION_SQL = """
+ALTER TABLE sharded_events
+ON CLUSTER '{cluster}'
+MATERIALIZE PROJECTION events_inserted_at_projection;
```
Comment (fuziontech) [inline posthog/clickhouse/migrations/0049_add_inserted_at_projection.py:15]: Could we not materialize? This seems like something that will only really be important for data that is showing up in the future?

I don't have the queries that we are using here to reference, but I imagine that most of them are for this current partition (this month). We could just apply this projection to the current partition by adding `IN PARTITION <partition_name>`

```suggestion
MATERIALIZE_PROJECTION_SQL = """
ALTER TABLE sharded_events
ON CLUSTER '{cluster}'
MATERIALIZE PROJECTION events_inserted_at_projection
IN PARTITION '202309';
```

Comment (tomasfarias) [reply posthog/clickhouse/migrations/0049_add_inserted_at_projection.py:15]: Yeah, I think this should be fine to do, thanks for the suggestion

Comment (tomasfarias) [reply posthog/clickhouse/migrations/0049_add_inserted_at_projection.py]: Worst case we can materialize more later in an async fashion?

**Comment by eboswort** [comment]
🙏 this MR would make a big difference for me and my team!

**Comment by eboswort** [comment]
Hi @tomasfarias, would you be able to help get this MR across the finish line? Really looking forward to being able to successfully sync all my posthog data to snowflake without missing stuff.

**Review by fuziontech — APPROVED** [review]
looks good to me! thanks for only doing this on `202309` partition

**Comment by fuziontech** [comment]
if you wanted to limit this to `202310` I would be even happier tbh!

**Comment by tomasfarias** [comment]
> if you wanted to limit this to `202310` I would be even happier tbh!

Updated the materialization to `202310`. Will merge if tests are green, thank you.

**Pull request merged**